File: Leetcode/Competition/1.py
class Solution:
    def generateTheString(self, n: int) -> str:
        if n & 1 == 0:
            ans = ''.join(['a'*(n-1),'b'])

        else:
            ans = 'a'*n
        return ans
            
print(Solution().generateTheString(7))

File: Leetcode/Competition/LCCUP20/1.cpp
#include <string>
using namespace std;

class Solution {
public:
    int calculate(string s) {
        int x = 1, y = 0;
        for (auto c : s){
            if(c == 'A'){
                x = 2 * x + y;
            }else if (c == 'B'){
                y = 2 * y + x;
            }
        }
        return x + y;
    }
};